src/pages/home/report/ReportActionCompose.js

```diff
@@ -177,8 +177,12 @@ class ReportActionCompose extends React.Component {
                                     )}
                                 </AttachmentPicker>
                                 <TextInputFocusable
+                                    noFocus={this.props.isSidebarShown && this.props.isSmallDevice}
                                     multiline
-                                    ref={el => this.textInput = el}
+                                    ref={el => {
+                                        this.textInput = el;
+                                        this.textInput.setNativeProps({tabindex: '-1'});
```

Comment:
What is the advantage of using `setNativeProps` here? `TextInputFocusable` has a web version here so we could maybe set `tabIndex: -1` there?

https://github.com/Expensify/Expensify.cash/blob/5037950db4c991bf2c433aaac01a5adea01910d3/src/components/TextInputFocusable/index.js#L190-L192

However the more I look into this solution... the less sure I am that agree this input should _not be focusable_ at small window widths. I'd expect the behavior to be that the sidebar closes when we tab over to the input. 

cc @roryabraham thoughts?

Reply:
@marcaaron 
The advantage of using `setNativeProps` here is that we want the `tabindex `to only effect this instance of the component, not everywhere in other usages.
As for the tab behavior if we treat the small device version as an app then pressing the arrow down after the initial screen which is the sidebar would have a strange effect that takes you to the TextInputFocusable in a conversation that wasn't tapped by the user in the first place.

Reply:
>we want the tabindex to only effect this instance of the component, not everywhere in other usages.

Yes, but I'm unsure why we want to do this inside the `ReportActionCompose`. We are already passing a prop of `isFocusable`. So can we use that prop within `TextInputFocusable` instead of calling `setNativeProps()` here?

Reply:
@marcaaron 
Because **isFocusable** and **tabindex** are two different things, maybe another developer wants an input **not** to be **focusable** on mount but wants the **tabindex** to be **available** in a form, therefore as a general case it would be better to separate both.
However if this is not the case then I can change it.

Reply:
>maybe another developer wants an input not to be focusable on mount but wants the tabindex to be available in a form

I'm not sure who this developer is, but if they asked for set `isFocusable` to `false` then why can we not assume that:

- The element would have a `tabIndex` of `-1`
- When attempting to focus it programmatically it would not focus

If we just set this input to have a `tabIndex` of `-1` then no user will be able to tab to it at all. Why prevent this when the element is in view? That's why I am suggesting that we use the `isFocusable` prop to determine _when_ to set the `tabIndex`.

Reply:
I think the naming `isFocusable` is misleading, I think the naming should be `isFocusedOnMount` because the intended behavior is to focus on the input on mount or not to fix a problem where the input being focused on mobile phones behind the slider when the slider is opened, therefore tabIndex has nothing to do with `isFocusable`, then what I suggest is the following:
- Changing the `isFocusable `to `isFocusedOnMount`
- Moving `tabIndex` of `-1` to inside `TextInputFocusable` and control it with an optional prop named `isSwitchable`.

Reply:
Ok, how about instead of `isFocusedOnMount` we call it `shouldFocusOnInit`? And also just allow passing in the `tabIndex` as a prop rather than introduce a concept of "switchable". Would that work?